fabric-object-builder-api-v1/src/main/java/net/fabricmc/fabric/impl/object/builder/TradeOfferInternals.java

```diff
@@ -63,4 +92,53 @@ public static void printRefreshOffersWarning() {
 		Throwable loggingThrowable = new Throwable();
 		LOGGER.warn("TradeOfferHelper#refreshOffers does not do anything, yet it was called! Stack trace:", loggingThrowable);
 	}
+
+	public static class WanderingTraderOffersBuilderImpl implements TradeOfferHelper.WanderingTraderOffersBuilder {
+		static final Object2IntMap<Identifier> ID_TO_INDEX = Util.make(new Object2IntOpenHashMap<>(), idToIndex -> {
+			idToIndex.put(BUY_ITEMS_POOL, 0);
+			idToIndex.put(SELL_SPECIAL_ITEMS_POOL, 1);
+			idToIndex.put(SELL_COMMON_ITEMS_POOL, 2);
+		});
+
+		/**
+		 * Make the trade list modifiable.
+		 */
+		static void initWanderingTraderTrades() {
+			if (!(TradeOffers.REBALANCED_WANDERING_TRADER_TRADES instanceof ArrayList)) {
+				TradeOffers.REBALANCED_WANDERING_TRADER_TRADES = new ArrayList<>(TradeOffers.REBALANCED_WANDERING_TRADER_TRADES);
+			}
+		}
+
+		@Override
+		public TradeOfferHelper.WanderingTraderOffersBuilder pool(Identifier id, int count, TradeOffers.Factory... factories) {
+			if (factories.length == 0) throw new IllegalArgumentException("cannot add empty pool");
+			if (count <= 0) throw new IllegalArgumentException("count must be positive");
+
+			Objects.requireNonNull(id, "id cannot be null");
+			Pair<TradeOffers.Factory[], Integer> pool = Pair.of(factories, count);
+			initWanderingTraderTrades();
+			ID_TO_INDEX.put(id, TradeOffers.REBALANCED_WANDERING_TRADER_TRADES.size());
+			TradeOffers.REBALANCED_WANDERING_TRADER_TRADES.add(pool);
+			return this;
+		}
+
+		@Override
+		public TradeOfferHelper.WanderingTraderOffersBuilder addOffersToPool(Identifier pool, TradeOffers.Factory... factories) {
+			if (!ID_TO_INDEX.containsKey(pool)) {
+				throw new IllegalArgumentException("pool %s is not registered".formatted(pool));
+			}
```

Comment:
How does adding offers to modded pools work? From what I can tell this depends on mod order.
Should probably do the additions later (also a supplier maybe?).

Reply:
Yeah, I suppose it's better to handle it ourselves.